kong/runloop/handler.lua

```diff
@@ -1309,13 +1309,24 @@ return {
 
           local upstream_scheme = var.upstream_scheme
           if upstream_scheme == "http"  and balancer_data.port ~= 80 or
-             upstream_scheme == "https" and balancer_data.port ~= 443
+             upstream_scheme == "https" and balancer_data.port ~= 443 or
+             upstream_scheme == "grpc"  and balancer_data.port ~= 80 or
+             upstream_scheme == "grpcs" and balancer_data.port ~= 443
           then
             upstream_host = upstream_host .. ":" .. balancer_data.port
           end
 
           var.upstream_host = upstream_host
         end
+
+        -- the nginx grpc module does not offer a way to overrride
+        -- the :authority pseudo-header; use our internal API to
+        -- do so
+        local resty_grpc = require "resty.kong.grpc"
```

Comment:
We should add a simple new PDK function that wraps around the module API (e.g. https://github.com/Kong/kong/blob/master/kong/pdk/service.lua#L136).

In fact, this file currently contains a few calls to the `resty.kong` Lua APIs, but they are all wrapped by PDKs.